docs/getting-started-guides/scratch.md

```diff
@@ -0,0 +1,727 @@
+  is not strictly necessary.
+- Nodes will need to run some version of Linux with the x86_64 architecture.  It may be possible
+  to run on other OSes and Architectures, but this guide does not try to assist with that.
+- Apiserver and etcd together are fine on a machine with 1 core and 1GB RAM for clusters with 10s of nodes.
+  Larger or more active clusters may benefit from a more cores.
+- Other nodes can have any reasonable amount of memory and any number of cores.  They need not
+  have identical configurations.
+
+### Network
+Kubernetes has a distinctive [networking model](../networking.md).
+
+Kubernetes allocates an IP address to each pod and creates a virtual ethernet device for each
+Pod.  When creating a cluster, you need to allocate a block of IPs that Kubernetes to be used
+as Pod IPs.  The normal approach is to allocate a different block to each node in the cluster
+as the node is added.  A process in one pod should be able to communicate with another pod
+using the IP of the second pod.  This connectivity can be accomplished in two ways:
+- Configure network to route Pod IPs
+  - Harder to setup from scratch.
+  - The Google Compute Engine ([GCE](gce.md)) and [AWS](aws.md) guides use this approach.
+  - Need to make the Pod IPs routable by programming routers, switches, etc.
+  - Can be configured external to kubernetes, or can implement in the "Routes" interface of a Cloud Provider module.
+- Create an Overlay network
+  - Easier to setup
+  - Traffic is encapsulated, so per-pod IPs are routable.
+  - Examples:
+    - [Flannel](https://github.com/coreos/flannel) 
+    - [Weave](http://weave.works/)
+  - Does not require "Routes" portion of Cloud Provider module.
+
+You need to select an address range for the virtual IPs. 
+- Various approaches:
+  - GCE: each project has own `10.0.0.0/8`.  Carve off a `/16` from that.  Room for several clusters in there.
+  - AWS: use one VPC for whole organization, carve off a chunk for each cluster.  Or use different VPC for different clusters.
+  - IPv6 not supported yet.
+- Allocate one CIDR for PodIPs for each node, or a large CIDR from which
+  smaller CIDRs are automatically allocated to each node (if nodes are dynamically added).
+  - You need Max-pods-per-node * max-number-of-nodes-expected IPs.
+  - e.g. use 10.240.0.0/16
+  - Need to make these routable or connect with overlay.
+
+Kubernetes also allocates an IP to each [service](../services.md).  However, service IPs do not necessarily
+need to be routable.  The kube-proxy takes care of translating Service IPs to Pod IPs before traffic leaves
+the node.  You do need to Allocate a block of IPs for services.  Call this `SERVICE_CLUSTER_IP_RANGE`.
+ Example: `SERVICE_CLUSTER_IP_RANGE="10.0.1.0/24"`
+
+Also, you need to pick a static IP for master node.
+- Call this `MASTER_IP`.
+- Open any firewalls to allow access to the apiserver ports 80 and/or 443.
```

Comment:
Seems like docker does this for you now? https://docs.docker.com/articles/networking/ says the default value of `--ip-forward` is true and when true docker sets this sysctl?